flexmeasures/data/models/planning/storage.py

```diff
@@ -241,20 +241,23 @@ def _prepare(self, skip_validation: bool = False) -> tuple:  # noqa: C901
         all_usage_forecast = []
 
         for component in usage_forecast:
-            all_usage_forecast.append(
-                get_continous_series_sensor_or_quantity(
-                    quantity_or_sensor=component,
-                    actuator=sensor,
-                    target_unit="MWh",
-                    query_window=(start, end),
-                    resolution=resolution,
-                    beliefs_before=belief_time,
-                )
+            gain = get_continous_series_sensor_or_quantity(
+                quantity_or_sensor=component,
+                actuator=sensor,
+                target_unit="MWh",
+                query_window=(start, end),
+                resolution=resolution,
+                beliefs_before=belief_time,
             )
+
+            if isinstance(component, Sensor):
+                gain *= resolution / component.event_resolution
```

Comment:
This is is because the usage forecast is defined in energy units per time period. For this reason, we assume that the constant usage forecast is already defined in the context of the resolution. We could assume an incoming resolution but this would be equivalent to defining a power. 

The main reason to have energy units is to 

NB: the `QuantityOrSensor` field only accepts one unit .

Reply:
Your answer could use a little touchup (unfinished sentence?), but I think I get the jist of it.

I just realised that the field has MWh units. Wouldn't MW make more sense, especially if the usage forecast is passed as a constant quantity? Otherwise it's up to interpretation how a certain amount of MWh would be allocated over time (over each event of the scheduled power sensor, or over each event of the usage sensor, or over the planning horizon).

Reply:
Indeed, MW suits more here. I've switched the units from energy to power in the last commit.